cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTableRegistry.java

```diff
@@ -71,7 +73,18 @@ public NoSqlStructuredTableRegistry(
     this.tableDefinition = tableDefinition;
     this.entityRegistrySpec = tableDefinition.configure(ENTITY_REGISTRY, DatasetProperties.EMPTY);
     this.transactional = Transactions.createTransactionalWithRetry(
-      NoSqlTransactionRunner.createTransactional(txClient, this::getRegistryTable),
+      NoSQLTransactionals.createTransactional(
+        txClient,
+        new TableDatasetSupplier() {
+          @Override
+          public <T extends Dataset> T getTableDataset(String name) throws IOException {
+            if (ENTITY_REGISTRY.equals(name)) {
```

Comment:
I just found that the naming convention of this class is quite inconsistent and confusing. Both `ENTITY_REGISTRY` and `REGISTRY_DATASET_ID` are actually referring to the same thing. In fact, looking the usage of `REGISTRY_DATASET_ID` of this class, it is always only need the name part. So why bother defining the `REGISTRY_DATASET_ID` at all?

Reply:
Removed `REGISTRY_DATASET_ID`